api/filter_dimension_options.go

```diff
@@ -149,46 +161,84 @@ func (api *FilterAPI) getFilterBlueprintDimensionOptionHandler(w http.ResponseWr
 		return
 	}
 
+	bytes, err := json.Marshal(dimensionOption)
+	if err != nil {
+		log.ErrorC("failed to marshal filter blueprint dimension option into bytes", err, logData)
+		if auditErr := api.auditor.Record(r.Context(), getOptionAction, actionUnsuccessful, auditParams); auditErr != nil {
+			handleAuditingFailure(r.Context(), getOptionAction, actionUnsuccessful, w, auditErr, logData)
+			return
+		}
+		http.Error(w, internalError, http.StatusInternalServerError)
+		return
+	}
+
 	if auditErr := api.auditor.Record(r.Context(), getOptionAction, actionSuccessful, auditParams); auditErr != nil {
 		handleAuditingFailure(r.Context(), getOptionAction, actionSuccessful, w, auditErr, logData)
 		return
 	}
 
 	setJSONContentType(w)
-	w.WriteHeader(http.StatusNoContent)
+	w.WriteHeader(http.StatusOK)
+	_, err = w.Write(bytes)
+	if err != nil {
+		log.ErrorC("failed to write bytes for http response", err, logData)
+		setErrorCode(w, err)
+		return
+	}
 
 	log.Info("got dimension option for filter blueprint", logData)
```

Comment:
Again replace all `log.Info(...)` or `log.InfoC(...)` with `log.InfoCtx(r.Context(), <string/description>, <log data>)`

If you have used any other log levels such as Trace or Debug, there should be changed to Info as expected by our logging standards.